testing_utilities.go

```diff
@@ -61,8 +66,6 @@ func compareReaders(t *testing.T, reader1 ion.Reader, reader2 ion.Reader) {
 			assert.NoError(t, reader1.StepOut(), "Something went wrong executing reader1.StepOut()")
 
 			assert.NoError(t, reader2.StepOut(), "Something went wrong executing reader2.StepOut()")
-		default:
-			t.Error(&InvalidIonTypeError{reader1.Type()})
 		}
```

Comment:
I would leave the default it with an error in case something is added/changed to make the code fail. If you believe the default should never be taken add a comment as to why.

Reply:
Done.